pkg/kubelet/dockertools/docker.go

```diff
@@ -1049,3 +1049,21 @@ func RunContainer(client DockerInterface, container *api.Container, pod *api.Pod
 	}
 	return dockerContainer.ID, nil
 }
+
+// KillContainer kills a docker container with the given containerID.
+// TODO(yifan): To use strong type for the containerID.
+func KillContainer(client DockerInterface, containerID string,
```

Comment:
Hmmm I'm not sure I like how this is looking. It seems like all the readiness and event recording should be outside of the runtime. It is conceptually all work Kubelet does outside of what container runtime it uses. Do you have a plan on how readiness, ref, and recorder will be used in the runtime? Seems like Kubelet can call KillContainer (which kills) and can do teh recording and readiness itself from the status.